LocalSettings.php

```diff
@@ -305,34 +327,56 @@
+$wgGroupPermissions['oversight']['blockemail'] = true;
 $wgGroupPermissions['oversight']['abusefilter-hide-log'] = true;
 $wgGroupPermissions['oversight']['abusefilter-hidden-log'] = true;
+$wgGroupPermissions['oversight']['abusefilter-log-detail'] = true;
+$wgGroupPermissions['oversight']['abusefilter-view-private'] = true;
+$wgGroupPermissions['oversight']['abusefilter-log-private'] = true;
+$wgGroupPermissions['oversight']['browsearchive'] = true;
+$wgGroupPermissions['oversight']['deletedhistory'] = true;
+$wgGroupPermissions['oversight']['deletedtext'] = true;
 $wgRevokePermissions['exampleuser']['editmyoptions'] = true;
 $wgRevokePermissions['exampleuser']['editmyprivateinfo'] = true;
-$wgAddGroups['sysop'] = array( 'bot', 'editor', 'reviewer', 'autoreview', 'patroller', 'chatmod', 'blockedfromchat' );
-$wgRemoveGroups['sysop'] = array( 'bot', 'editor', 'reviewer', 'autoreview', 'patroller', 'chatmod', 'blockedfromchat' );
-$wgAddGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'exampleuser', 'oversight' );
-$wgRemoveGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'exampleuser', 'oversight' );
-$wgAddGroups['bureaucrat'] = array( 'sysop', 'bureaucrat' );
-$wgRemoveGroups['bureaucrat'] = array( 'sysop', 'bureaucrat' );
+$wgGroupPermissions['researcher']['browsearchive'] = true;
+$wgGroupPermissions['researcher']['deletedhistory'] = true;
+$wgGroupPermissions['researcher']['deletedtext'] = true;
+$wgGroupPermissions['chatmod']['chat'] = false;
+$wgGroupPermissions['chatmod']['viewpmlog'] = true;
+$wgGroupPermissions['reviewer']['validate'] = true;
+$wgGroupPermissions['reviewer']['review'] = true;
+$wgGroupPermissions['reviewer']['autoreview'] = true;
+$wgGroupPermissions['reviewer']['autoconfirmed'] = true;
+$wgGroupPermissions['reviewer']['editsemiprotected'] = true;
+$wgGroupPermissions['reviewer']['unreviewedpages'] = true;
+$wgGroupPermissions['editor']['review'] = true;
+$wgGroupPermissions['editor']['autoreview'] = true;
+$wgGroupPermissions['editor']['autoconfirmed'] = true;
+$wgGroupPermissions['editor']['editsemiprotected'] = true;
+$wgGroupPermissions['editor']['unreviewedpages'] = true;
+$wgGroupPermissions['autopatrol']['autopatrol'] = true;
+$wgAddGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 'researcher', 'chatmod', 'blockedfromchat' );
+$wgRemoveGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 'researcher', 'chatmod', 'blockedfromchat' );
+$wgAddGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'oversight' );
+$wgRemoveGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'oversight' );
+$wgAddGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat' );
+$wgRemoveGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat' );
+$wgGroupsRemoveFromSelf = [ '*' => true ];
```

Comment:
What is this even supposed to do?

Reply:
$wgGroupsRemoveFromSelf = [ '*' => true ];
will allow anyone to remove themselves (and only themselves) from any group that happen to be in. It basically enables self demotion.